version: '3.8'
services:
  user_db:
    container_name: user_db
    image: mysql:latest
    networks:
      - auth_network
    ports:
      - "3306:3306"
    volumes:
      - auth_user_db_log:/var/lib/mysql
    env_file:
      - .env.user.db
  session_db:
    container_name: session_db
    image: mysql:latest
    networks:
      - auth_network
    ports:
      - "3307:3306"
    volumes:
      - auth_session_db_log:/var/lib/mysql
    env_file:
      - .env.session.db
  redis_db:
    container_name: redis_db
    image: redis:alpine
    networks:
      - auth_network
    ports:
      - "6379:6379"
    command: redis-server /usr/local/etc/redis/redis.conf
    volumes:
      - auth_redis_db_log:/data
      - .redis.conf:/usr/local/etc/redis/redis.conf
    environment:
      - TZ=Asia/Seoul
  register_server:
    container_name: register_server
    build: ./register_server
    networks:
      - auth_network
    ports:
      - "8000:8000"
    volumes:
      - auth_register_app_log:/var/log
    env_file:
      - ./register_server/register_app/.env
    develop:
        watch:
          - action: sync+restart
            path: ./register_server
            target: /app
            ignore:
            - README.md
  auth_server:
    container_name: auth_server
    build: ./auth_server
    networks:
      - auth_network
    ports:
      - "8002:8000"
    volumes:
      - auth_app_log:/var/log
    env_file:
      - ./auth_server/auth_app/.env
    develop:
        watch:
          - action: sync+restart
            path: ./auth_server
            target: /app
            ignore:
            - README.md

volumes:
  auth_register_app_log:
    external: true
  auth_user_db_log:
    external: true
  auth_session_db_log:
    external: true
  auth_redis_db_log:
    external: true
  auth_app_log:
    external: true

networks:
  auth_network:
    external: true